### DISCUSSION TITLE
API Documentation or complete example with all methods?

### DISCUSSION THREAD

**Discussion by baj1210** [ghiscoding/slickgrid-react#425] [Ideas]
I understand this is a labor of love and writing documentation is not much fun. This project has the potential to be very powerful, but I'm really struggling to get things to work. I would almost prefer one complete example that covers everything (editing, graphql backend, resizing, tooltips, etc. with the plugin intergations needed to get these things to work) instead of 50 individual demo examples that do a good job showing capabilities, but are less effective as a learning tool. 

If I were more skilled, I would write this myself and contribute. Currently, my understanding of it is too poor and I am guessing and trying to figure out the various options and what works/what doesn't etc the painful way.

**Comment by ghiscoding** [comment]
Well sorry but I seriously have no interest in helping someone who is coming in here just to complain on an Open Source project that is provided to him at no cost (free) and I also don't think that you have put much effort into trying the project and reading the documentations either. I already spent countless hours in this project over the years without being paid, so reading comments like yours is not something that I consider very helpful (you should think twice before posting such comments). 

The reasons that there are so many examples is that they all represent different features which are better represented as separate examples for most users (you excluded apparently) and it is also much better for the project to implement more separate E2E testing (Cypress) which I have in place to make sure that the project is stable.  The Example 3 with Editing is typically what most developers would use or at least start with, so you should study that one to get started.

I also believe that you might have missed the ton of available [Documentations](https://ghiscoding.gitbook.io/slickgrid-react/getting-started/quick-start) that I created over the years and the link to these docs is provided in multiple areas (on the main homepage readme and also in the live demo). I also created the [Slickgrid-React-Demos](https://github.com/ghiscoding/slickgrid-react-demos) that you can clone and try for yourself, it is also updated on every new releases. I seriously don't understand why you're complaining when there are so many documentations and examples provided freely. 

Side note I am **not** a React developer and I also have no intention to be either, this project was created by someone else and was later transferred to my name because I support a bunch of other frameworks related to SlickGrid (including this one since then). Whatever questions you have, I think that you should ask them on Stack Overflow which is a much better platform to ask questions (especially if it's React related). 

This is an open source library that is provided "as is" and is not intended to be a course, a tutorial or any of that. If you still think this is not good enough for you, well sorry but I think you should just look elsewhere, there are tons of React [data grid libraries](https://reactjsexample.com/20-best-react-table-components-and-libraries/) or you could simply try [Ag-Grid](https://www.ag-grid.com/) which has superb documentations and comes with full developer support but it also comes with a very hefty price.

**Comment by baj1210** [comment]
It's not complaining, just a suggestion. Maybe there's a way to
autodocument in javascript? I have no idea. Maybe supporting so many
projects at once is causing stress?

**Reply by ghiscoding** [reply]
That's why TypeScript is so useful, it provides intellisense and JavaScript doesn't always. Also like I said, I have plenty of Documentations, if any of them needs an update then they are all markdown files that can be found in the [docs](https://github.com/ghiscoding/slickgrid-react/tree/master/docs) folder which automatically generate the [Documentations](https://ghiscoding.gitbook.io/slickgrid-react/) of the project.
